lib/src/themes/theme_function.dart

```diff
@@ -1,23 +1,30 @@
 import 'dart:math';
 import 'dart:ui';
 
+import 'package:vector_tile_renderer/src/expressions/expression.dart';
+import 'package:vector_tile_renderer/src/expressions/value_expression.dart';
+
 import 'theme_function_model.dart';
 
 abstract class ThemeFunction<T> {
   Map<FunctionModel, _ZoomValue> _cache = {};
 
-  T? exponential(FunctionModel<T> model, double zoom) {
+  T? exponential(FunctionModel<T> model, Map<String, dynamic> args) {
+    final zoom = args['zoom'] as double;
```

Comment:
now that we're passing `args` and not just a zoom, the cached value could be incorrect depending on the arguments used by the function, right? in other words, is this cache now broken?

The idea behind the cache is to optimize rendering, i.e. only calculate expensive expressions once when rendering. I found that this made a big difference in the profiler, since tiles can result in evaluating expressions 1000s of times, and some expressions are expensive.

Reply:
This is a good point and honestly, I am not sure. The arguments passed to the function are just the properties of the VectorTileFeature that's currently being rendered which by my understanding stay identical per feature regardless of the zoom making the zoom level the only relevant property from the args for caching. I am just uncertain whether my assumption of the properties of VectorTileFeatures remaining identical is accurate or not (if it was accurate, maybe it's still enough because the properties are still tied to the zoom level meaning the zoom level would still be sufficient as cache key).

What's your take on this?

Reply:
If the properties of the VectorTileFeature are used in calculations, then this needs to be reworked since the cache would be broken. If they're not used in calculations, then they should not be passed in because it could lead to future bugs and makes the code more complicated.